nomad/structs/structs.go

```diff
@@ -5366,9 +5368,14 @@ func (d *ParameterizedJobConfig) Copy() *ParameterizedJobConfig {
 
 // DispatchedID returns an ID appropriate for a job dispatched against a
 // particular parameterized job
-func DispatchedID(templateID string, t time.Time) string {
+func DispatchedID(prefix, templateID string, t time.Time) string {
 	u := uuid.Generate()[:8]
-	return fmt.Sprintf("%s%s%d-%s", templateID, DispatchLaunchSuffix, t.Unix(), u)
+
+	if prefix == "" {
+		prefix = fmt.Sprint(t.Unix())
+	}
```

Comment:
Getting rid of the unix timestamp entirely means that these won't sort by name anymore.